providers/aws/resources/aws_waf.go

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package resources
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/wafv2"
+	waftypes "github.com/aws/aws-sdk-go-v2/service/wafv2/types"
+
+	//"github.com/aws/aws-sdk-go/aws"
+	"go.mondoo.com/cnquery/v9/llx"
+	"go.mondoo.com/cnquery/v9/providers-sdk/v1/util/convert"
+	"go.mondoo.com/cnquery/v9/providers/aws/connection"
+	"go.mondoo.com/cnquery/v9/types"
+)
+
+func (a *mqlAwsWaf) id() (string, error) {
+	return "aws.waf", nil
+}
+
+func (a *mqlAwsWafAcl) id() (string, error) {
+	return a.Id.Data, nil
+}
+
+func (a *mqlAwsWafRule) id() (string, error) {
+	return a.Name.Data, nil
+}
+
+func (a *mqlAwsWafRulegroup) id() (string, error) {
+	return a.Name.Data, nil
+}
+
+func (a *mqlAwsWafIpset) id() (string, error) {
+	return a.Name.Data, nil
+}
+
+func (a *mqlAwsWaf) acls() ([]interface{}, error) {
+	conn := a.MqlRuntime.Connection.(*connection.AwsConnection)
+	//waf := a.Id.Data
+
+	region := ""
+	svc := conn.Wafv2(region)
+	ctx := context.Background()
+	acls := []interface{}{}
+	//scope := "REGIONAL"
+	nextMarker := aws.String("No-Marker-to-begin-with")
+	var scope waftypes.Scope
+	scope = "REGIONAL"
+	params := &wafv2.ListWebACLsInput{Scope: scope}
+	for nextMarker != nil {
+		aclsRes, err := svc.ListWebACLs(ctx, params)
+		if err != nil {
+			return nil, err
+		}
+		nextMarker = aclsRes.NextMarker
+		if aclsRes.NextMarker != nil {
+			params.NextMarker = nextMarker
+		}
+
+		for _, acl := range aclsRes.WebACLs {
+			params := &wafv2.GetWebACLInput{
+				Id:    acl.Id,
+				Name:  acl.Name,
+				Scope: scope,
+			}
+			aclDetails, err := svc.GetWebACL(ctx, params)
+			if err != nil {
+				return nil, err
+			}
+			mqlAcl, err := CreateResource(a.MqlRuntime, "aws.waf.acl",
+				map[string]*llx.RawData{
+					"id":                       llx.StringDataPtr(acl.Id),
+					"arn":                      llx.StringDataPtr(acl.ARN),
+					"name":                     llx.StringDataPtr(acl.Name),
+					"description":              llx.StringDataPtr(acl.Description),
+					"managedByFirewallManager": llx.BoolData(aclDetails.WebACL.ManagedByFirewallManager),
+				},
+			)
+			if err != nil {
+				return nil, err
+			}
+			acls = append(acls, mqlAcl)
+		}
+	}
+	return acls, nil
+}
+
+func (a *mqlAwsWafAcl) rules() ([]interface{}, error) {
+	conn := a.MqlRuntime.Connection.(*connection.AwsConnection)
+	//waf := a.Id.Data
+
+	var scope waftypes.Scope
+	scope = "REGIONAL"
+	ctx := context.Background()
+	region := ""
+	svc := conn.Wafv2(region)
+	rules := []interface{}{}
+	params := &wafv2.GetWebACLInput{
+		Id:    &a.Id.Data,
+		Name:  &a.Name.Data,
+		Scope: scope,
+	}
+	aclDetails, err := svc.GetWebACL(ctx, params)
+	if err != nil {
+		return nil, err
+	}
+	for _, rule := range aclDetails.WebACL.Rules {
+		ruleStatement, err := convert.JsonToDict(rule.Statement)
```

Comment:
marius and i chatted about this - the top level items under statement should be objects instead of dict to make querying easier